src/flood_wave_detector.py

```diff
@@ -71,10 +72,48 @@ def run(self) -> None:
         Executes the steps needed to find all the flood waves.
         """
         self.mkdirs()
-        self.find_vertices()
-        self.find_edges()
+
+        stations_life_intervals = JsonHelper.read(filepath=os.path.join(PROJECT_PATH,
+                                                                        'data', 'existing_stations.json'))
+        all_dates = [self.start_date, self.end_date]
+        for gauge in self.gauges:
+            x = stations_life_intervals[str(gauge)]["start"]
+            y = stations_life_intervals[str(gauge)]["end"]
+            all_dates = self.add_elem(list0=all_dates, date=x)
+            all_dates = self.add_elem(list0=all_dates, date=y)
+        all_dates.sort(key=lambda d: datetime.strptime(d, '%Y-%m-%d'))
+        cut_dates = [date for date in all_dates if self.start_date <= date <= self.end_date]
+
+        for i in range(len(cut_dates) - 1):
+            gauges_copy = self.gauges
+
+            self.start_date = cut_dates[i]
+            self.end_date = cut_dates[i + 1]
+
+            exist = str(datetime.strptime(cut_dates[i], "%Y-%m-%d") + timedelta(days=1))
+            existing_gauges = []
+            for gauge in self.gauges:
+                if stations_life_intervals[str(gauge)]["start"] <= exist <= stations_life_intervals[str(gauge)]["end"]:
+                    existing_gauges.append(gauge)
+            self.gauges = existing_gauges
+
+            self.find_vertices()
+            self.find_edges()
+
+            # Set original values
+            self.gauges = gauges_copy
+            self.start_date = cut_dates[0]
+            self.end_date = cut_dates[-1]
+
         GraphBuilder().build_graph(folder_name=self.folder_name)
 
+    @staticmethod
+    def add_elem(list0: list, date: str):
```

Comment:
You can remove this method and replace its call with the syntax: `list0 + [date] if date not in list0 else list0`